File: examples/Lecture7/RoomPersistenceLibraryDemo/app/src/main/java/mobiledev/unb/ca/roompersistencetest/DBManager.java
package mobiledev.unb.ca.roompersistencetest;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.SQLException;
import android.database.sqlite.SQLiteDatabase;

public class DBManager {
    private DatabaseHelper dbHelper;
    private Context context;
    private SQLiteDatabase database;

    public DBManager(Context context) {
        this.context = context;
    }

    public DBManager open() throws SQLException {
        dbHelper = new DatabaseHelper(context);
        database = dbHelper.getReadableDatabase();
        return this;
    }

    public void close() {
        dbHelper.close();
    }

    public void insertRecord(String item, String num) {
        ContentValues contentValue = new ContentValues();
        contentValue.put(DatabaseHelper.ITEM, item);
        contentValue.put(DatabaseHelper.NUM, num);
        database.insert(DatabaseHelper.TABLE_NAME, null, contentValue);
    }

    public Cursor listAllRecords() {
        Cursor cursor = database.query(DatabaseHelper.TABLE_NAME, DatabaseHelper.COLUMNS, null, null, null, null, null);
        if (cursor != null) {
            cursor.moveToFirst();
        }
        return cursor;
    }

    public void deleteRecord(String item) {
        database.delete(DatabaseHelper.TABLE_NAME,
                DatabaseHelper._ID + "=?",
                new String[]{item});
    }
}

File: Winter2022/examples/kotlin/Lecture 7/DBDemo/app/src/main/java/mobiledev/unb/ca/sqlitetest/DBManager.kt
package mobiledev.unb.ca.sqlitetest

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase

class DBManager(context: Context?) {
    private val dbHelper: DatabaseHelper = DatabaseHelper(context)
    fun close() {
        dbHelper.close()
    }

    fun listAllRecords(): Cursor? {
        val cursor = openReadOnlyDatabase().query(DatabaseHelper.TABLE_NAME,
            DatabaseHelper.COLUMNS,
            null,
            null,
            null,
            null,
            null)
        cursor?.moveToFirst()
        return cursor
    }

    fun insertRecord(item: String?, num: String?) {
        val contentValue = ContentValues()
        contentValue.put(DatabaseHelper.ITEM, item)
        contentValue.put(DatabaseHelper.NUM, num)
        openWriteDatabase().insert(DatabaseHelper.TABLE_NAME, null, contentValue)
    }

    fun deleteRecord(id: Int) {
        openWriteDatabase().delete(DatabaseHelper.TABLE_NAME,
            DatabaseHelper.ITEM_ID + "=?", arrayOf(id.toString()))
    }

    private fun openReadOnlyDatabase(): SQLiteDatabase {
        return dbHelper.readableDatabase
    }

    private fun openWriteDatabase(): SQLiteDatabase {
        return dbHelper.writableDatabase
    }

}